spec/lib/trace_client_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'spec_helper'
+
+module ZipkinTracer
+  describe TraceClient do
+    let(:key) { 'key' }
+    let(:value) { 'value' }
+    subject { TraceClient }
+
+    before do
+      allow(Trace).to receive(:default_endpoint).and_return(Trace::Endpoint.new('127.0.0.1', '80', 'service_name'))
+    end
+
+    describe '.record' do
+      it 'records an annotation' do
+        expect(Trace).to receive(:record).with(instance_of(Trace::Annotation))
+        subject.record(key)
+      end
+    end
+
+    describe '.record_tag' do
+      it 'records a binary annotation' do
+        expect(Trace).to receive(:record).with(instance_of(Trace::BinaryAnnotation))
+        subject.record_tag(key, value)
+      end
+    end
+
+    describe '.record_local_component' do
+      it 'records a binary annotation ' do
+        expect(Trace).to receive(:record).with(instance_of(Trace::BinaryAnnotation))
```

Comment:
maybe you can also check that the BinaryAnnotation got the proper parameters. I believe the specs of the faraday_middleware is doing that, you can copy it from there.